pkg/deploy/controller/image_change_controller.go

```diff
@@ -108,16 +135,26 @@ func parseImage(name string) (string, string) {
 	return name[:index], name[index+1:]
 }
 
-func generateTriggerDetails(triggers []deployapi.DeploymentTriggerImageChangeParams) *deployapi.DeploymentDetails {
+func generateTriggerDetails(config *deployapi.DeploymentConfig, triggers []deployapi.DeploymentTriggerImageChangeParams) *deployapi.DeploymentDetails {
 	// Generate the DeploymentCause objects from each DeploymentTriggerImageChangeParams object
 	// Using separate structs to ensure flexibility in the future if these structs need to diverge
 	causes := []*deployapi.DeploymentCause{}
 	for _, trigger := range triggers {
+		// The trigger From field takes precedence over RepositoryName
+		repoName := trigger.RepositoryName
+		if len(trigger.From.Name) > 0 {
+			namespace := trigger.From.Namespace
+			if len(namespace) == 0 {
+				namespace = config.Namespace
+			}
+			repoName = fmt.Sprintf("%s/%s", namespace, trigger.From.Name)
```

Comment:
You can't do this - you need to be loading this from ImageRepository.Status.ImageRepository.